db/testing.go

```diff
@@ -0,0 +1,53 @@
+package db
+
+import (
+	"fmt"
+
+	"github.com/attic-labs/noms/go/marshal"
+	"github.com/attic-labs/noms/go/types"
+	"github.com/attic-labs/noms/go/util/datetime"
+	"github.com/stretchr/testify/assert"
+	"roci.dev/diff-server/kv"
+)
+
+type testCommits []Commit
+
+func (t testCommits) genesis() Commit {
+	return t[0]
+}
+
+func (t testCommits) head() Commit{
+	return t[len(t)-1]
+}
+
+func (t *testCommits) addGenesis(assert *assert.Assertions, db *DB) *testCommits {
+	m := kv.NewMap(db.noms)
+	genesis := makeGenesis(db.noms, "", db.noms.WriteValue(m.NomsMap()), m.NomsChecksum(), 0)
+	_, err := db.noms.SetHead(db.noms.GetDataset(LOCAL_DATASET), db.noms.WriteValue(marshal.MustMarshal(db.noms, genesis)))
```

Comment:
```suggestion
	_, err := db.noms.SetHead(db.noms.GetDataset(LOCAL_DATASET), genesis.Original)
```

Unless you treat a `Commit` mutably (which you should not do), then there is no need to re-marshal it.